src/test/java/com/bravo/user/service/PaymentServiceTest.java

```diff
@@ -0,0 +1,107 @@
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import com.bravo.user.App;
+import com.bravo.user.dao.model.Payment;
+import com.bravo.user.dao.model.mapper.ResourceMapper;
+import com.bravo.user.dao.repository.PaymentRepository;
+import com.bravo.user.dao.specification.PaymentSpecification;
+import com.bravo.user.model.dto.PaymentDto;
+import com.bravo.user.utility.PageUtil;
+
+@ContextConfiguration(classes = { App.class })
+@ExtendWith(SpringExtension.class)
+@SpringBootTest
+class PaymentServiceTest {
+
+	@Autowired
+	private HttpServletResponse httpResponse;
+
+	@Autowired
+	private PaymentService paymentService;
+
+	@MockBean
+	private ResourceMapper resourceMapper;
+
+	@MockBean
+	private PaymentRepository paymentRepository;
+
+	private List<PaymentDto> dtoPayments;
+
+	@BeforeEach
+	public void beforeEach() {
+		final List<Integer> ids = IntStream.range(1, 10).boxed().collect(Collectors.toList());
+
+		final Page<Payment> mockPage = mock(Page.class);
+		when(paymentRepository.findAll(any(PaymentSpecification.class), any(PageRequest.class)))
+			.thenReturn(mockPage);
+
+
+		final List<Payment> daoPayments = ids.stream()
+			.map(id -> createPayment(Integer.toString(id)))
+			.collect(Collectors.toList());
+		when(mockPage.getContent()).thenReturn(daoPayments);
+		when(mockPage.getTotalPages()).thenReturn(9);
+
+		this.dtoPayments = ids.stream()
+			.map(id -> createPaymentDto(Integer.toString(id)))
+			.collect(Collectors.toList());
+		when(resourceMapper.convertPayments(daoPayments)).thenReturn(dtoPayments);
+	}
+
+	@Test
+	void retrieveByUserId() {
+		final String input = "input";
+		final PageRequest pageRequest = PageUtil.createPageRequest(null, null);
+		final List<PaymentDto> results = paymentService.retrieveByUserId(input, pageRequest, httpResponse);
+		assertEquals(dtoPayments, results);
+		assertEquals("9", httpResponse.getHeader("page-count"));
+		assertEquals("1", httpResponse.getHeader("page-number"));
+		assertEquals("20", httpResponse.getHeader("page-size"));
+	}
+
+	@Test
+	void retrieveByUserIdPaged() {
+		final String input = "input2";
+		final PageRequest pageRequest = PageUtil.createPageRequest(2, 5);
+		final List<PaymentDto> results = paymentService.retrieveByUserId(input, pageRequest, httpResponse);
+		assertEquals(dtoPayments, results);
+		assertEquals("9", httpResponse.getHeader("page-count"));
+		assertEquals("2", httpResponse.getHeader("page-number"));
+		assertEquals("5", httpResponse.getHeader("page-size"));
+	}
+
+	private Payment createPayment(final String id) {
+		final Payment payment = new Payment();
+		payment.setId(id);
+		return payment;
+	}
+
+	private PaymentDto createPaymentDto(final String id) {
+		final PaymentDto payment = new PaymentDto();
+		payment.setId(id);
+		return payment;
+	}
```

Comment:
This sets the payment id.  Are the other fields checked in these tests?

Reply:
Was mainly just following the structure of the UserServiceTest, but we could and should test other fields as well.

Reply:
The code in the repo isn't necessarily correct :) .  I have been following Uncle Bob pretty closely and was a bit shocked by his calling untested code "immoral" https://www.amazon.com/Clean-Coder-Conduct-Professional-Programmers/dp/0137081073.   While I think that is a bit overly harsh, from our business client's perspective it's of huge importance.  Thanks for the replies. I put in a good word about you to Issac. :)

Reply:
Great, thank you so much! And will have to check out that book and couldn't agree more that what matters for the client is what matters most.